coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
```

Comment:
Thats a weird edge case with deletions.

Assume a file like this: `file = ['a\n', 'b']`
We delete the last line, resulting in `file = ['a\n']`. 

Since the "eof-state" (i.e. no new line at the end of the file) should be kept, the `\n` now be deleted.
This would require functionality my current functions do not have, and would likely be more complicated to implement:
lots of checks like where did the deletion happen and did we have a `\n` at the eof or not, and probably also some other edge cases. 
Stripping all new lines and then just choosing the correct function, based on the original "eof-state" is far easier. 

Actually, all these edge cases between add, modify and delete led me to just changing the files (or diffs) we hand out from this function, after all changes have been applied.